pkg/server/app/app_test.go

```diff
@@ -0,0 +1,126 @@
+package app
+
+import (
+	"testing"
+
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/sqlite"
+	"github.com/luizalabs/teresa-api/models/storage"
+	"github.com/luizalabs/teresa-api/pkg/server/auth"
+	st "github.com/luizalabs/teresa-api/pkg/server/storage"
+)
+
+type fakeAppOperations struct{}
+
+type errAppOperations struct{ Err error }
+
+func (*fakeAppOperations) Create(app *App, st st.Storage) error {
+	return nil
+}
+
+func (e *errAppOperations) Create(app *App, st st.Storage) error {
+	return e.Err
+}
+
+func createFakeUser(db *gorm.DB, email string) (*storage.User, error) {
+	user := &storage.User{Email: email}
+	if err := db.Create(user).Error; err != nil {
+		return nil, err
+	}
+	return user, nil
+}
+
+func createFakeTeam(db *gorm.DB, name string) (*storage.Team, error) {
+	team := &storage.Team{Name: name}
+	if err := db.Create(team).Error; err != nil {
+		return nil, err
+	}
+	return team, nil
+}
+
+func addUserTeam(db *gorm.DB, user *storage.User, team *storage.Team) error {
+	return db.Model(team).Association("Users").Append(user).Error
+}
```

Comment:
What do you think about using `user.Operations` and `team.Operations` instead of `gorm` for those fake data builder?